pysindy/optimizers/trapping_sr3.py

```diff
@@ -479,16 +542,23 @@ def _solve_nonsparse_relax_and_split(self, H, xTy, P_transpose_A, coef_prev):
         return coef_sparse
 
     def _solve_direct_cvxpy(self, r, N, x_expanded, y, Pmatrix, coef_prev):
+        """
+        If using the direct formulation of trapping SINDy, solves the
+        entire problem in CVXPY regardless of the threshold value.
+        Note that this is a convex-composite (i.e. technically nonconvex)
+        problem, solved in CVXPY, so convergence/quality guarantees are
+        not available here!
+        """
         xi = cp.Variable(N * r)
         cost = cp.sum_squares(x_expanded @ xi - y.flatten())
         if self.thresholder.lower() == "l1":
             cost = cost + self.threshold * cp.norm1(xi)
-        elif self.thresholder.lower() == "weighted_l1":
-            cost = cost + cp.norm1(np.ravel(self.thresholds) * xi)
+        elif self.thresholder.lower() == "weighted_l1" and np.any(self.thresholds != 0):
```

Comment:
Why do we only do this additional check for the weighted L1 norm?

Reply:
As far as I can tell, this is unnecessary. The SR3 optimizer (which the TrappingSR3 optimizer derives from) already checks during initialization that if the thresholder is weighted, thresholds != 0. So I removed the second part of this if statement.